test/controllers/tasks/active.officers.controller.unit.ts

```diff
@@ -1,77 +1,117 @@
 
 describe("Active officers controller tests", () => {
 
   beforeEach(() => {
     mocks.mockAuthenticationMiddleware.mockClear();
+    mocks.mockServiceAvailabilityMiddleware.mockClear();
+    mocks.mockSessionMiddleware.mockClear();
+    mockGetActiveOfficerDetails.mockClear();
   });
 
-  it("Should return active officers page", async () => {
-    mockIsActiveFeature.mockReturnValueOnce(true);
-    mockGetCompanyOfficers.mockResolvedValueOnce(validCompanyOfficers);
-    const response = await request(app).get(url);
-    expect(response.text).toContain(EXPECTED_TEXT);
-  });
+  describe("get tests", () => {
 
-  it("Should return error page if active officers cannot be retired", async () => {
-    const message = "Can't connect";
-    mockGetCompanyOfficers.mockRejectedValueOnce(new Error(message));
-    mockIsActiveFeature.mockReturnValueOnce(true);
-    const response = await request(app).get(url);
-    expect(response.text).toContain(EXPECTED_ERROR_TEXT);
-  });
+    it("Should navigate to director's details page", async () => {
+      mockGetActiveOfficerDetails.mockResolvedValueOnce(mockActiveOfficerDetails);
+      const response = await request(app).get(ACTIVE_OFFICER_DETAILS_URL);
 
-  it("Should skip officer check and return active officers page if feature flag is off", async () => {
-    mockIsActiveFeature.mockReturnValueOnce(false);
-    const response = await request(app).get(url);
-    expect(response.text).toContain(EXPECTED_TEXT);
-  });
+      expect(response.text).toContain(PAGE_HEADING);
+      expect(response.text).toContain("Is the director still active?");
+      expect(response.text).toContain(mockActiveOfficerDetails.foreName1);
+      expect(response.text).toContain(mockActiveOfficerDetails.foreName2);
+      expect(response.text).toContain(mockActiveOfficerDetails.dateOfBirth);
+      expect(response.text).toContain(mockActiveOfficerDetails.nationality);
+    });
 
-  it("Should return to task list page when officer details is confirmed", async () => {
-    const response = await request(app).post(url).send({ activeDirectors: "yes" });
-    expect(response.status).toEqual(302);
-    expect(response.header.location).toEqual(TASK_LIST_URL);
-  });
+    it("Should navigate to director's details page with no middle name", async () => {
+      const fName2 = mockActiveOfficerDetails.foreName2;
+      mockActiveOfficerDetails.foreName2 = undefined;
+
+      mockGetActiveOfficerDetails.mockResolvedValueOnce(mockActiveOfficerDetails);
+      const response = await request(app).get(ACTIVE_OFFICER_DETAILS_URL);
+
+      expect(response.text).toContain(PAGE_HEADING);
+      expect(response.text).toContain("Is the director still active?");
+      expect(response.text).toContain(mockActiveOfficerDetails.foreName1);
+      expect(response.text).toContain(mockActiveOfficerDetails.dateOfBirth);
+      expect(response.text).toContain(mockActiveOfficerDetails.nationality);
+
```

Comment:
i think you just need one more expect(response.text).**not**.toContain(mockActiveOfficerDetails.foreName2);
just to check it is not on the page. That should be it then!